### ISSUE TITLE
May I ask if your COCO-Caption metrics were also obtained through two-stage training? What learning rate did you set for the second-stage fine-tuning?

### ISSUE THREAD

**Issue by geoffreyfan** [Graph-and-Geometric-Learning/MindLLM#2]

**Comment by Boltzmachine** [comment]
Yes. The results of COCO-caption are also fine-tuned. The learning rate is 1e-4. Note that, unlike Table 2, Table 1's baselines use another preprocessed version of NSD. (We re-process NSD since we need more information from the raw dataset). So you need to pre-train and finetune MindLLM on that version yourself. I plan to release that checkpoint in the future as well, but unfortunately, I do not have sufficient resources right now, given other ongoing projects.

And of course, you can try fine-tuning it with our current setting and our preprocessed version using the instruction in README; the score will be even higher.

<img width="909" height="618" alt="Image" src="https://github.com/user-attachments/assets/b15e3596-4bff-4106-b154-a013af2c9c1c" />

**Comment by geoffreyfan** [comment]
I see that your default settings are whole_brain: false and nsd_version: 0. May I ask when you would use the whole_brain option, as well as nsd_version: 1?

**Comment by Boltzmachine** [comment]
`whole_brain` will never be used. `nsd_version=1` means using MindEye's processing. But at least for now you need to figure out how to download and use the dataset yourself.

**Comment by geoffreyfan** [comment]
May I ask if in the first stage, the VRAM size significantly impacts the training effectiveness? I only have 80 GB of VRAM and trained the first stage with batch-size = 4, but the results were far from satisfactory.

**Comment by Boltzmachine** [comment]
Could you give me the complete hyperparameter list as well as the total training epochs/steps? would be better if you could share me with your wandb workspace

**Comment by geoffreyfan** [comment]
I used the default parameters from your public code, only changing the learning rate from 0.001 to 0.0001, and disabled wandb because it couldn't function properly on my university's servers. 

The validation loss stopped decreasing after epoch 5, and no subsequent epochs were saved. Below are the results from epoch 5:

<img width="848" height="1125" alt="Image" src="https://github.com/user-attachments/assets/bfc975d5-2045-4d9e-a60f-944632bef83b" />

**Comment by Boltzmachine** [comment]
Are you at the pretraining stage or the finetuning stage?

**Comment by geoffreyfan** [comment]
pretraining stage

**Comment by Boltzmachine** [comment]
Please strictly follow the command I write in README
```
python main.py group_by_coco=false "subjects=[1,2,3,4,5,6,7]"

```
to make sure you can reproduce. Apparently you modify the learning_rate and set `early_stop=true`.

**Comment by geoffreyfan** [comment]
I noticed you trained the first stage on eight GPUs for eight days, but I only have 80 GB of VRAM.

**Comment by Boltzmachine** [comment]
In that case I do not recommend you reproduce from the pretraining stage as it contains a large language model. Hardware resources would be a bottleneck for such a large model and there is no way to avoid it. But your resources would be enough for the finetuning stage.

**Comment by Boltzmachine** [comment]
unless you use that GPU to train it for 2 months

**Comment by geoffreyfan** [comment]
Okay, thank you very much.
